### DISCUSSION TITLE
Why is the TransactionEncoder() function transcoding my headers too?|为什么TransactionEncoder()函数会将我的标题也转码了？

### DISCUSSION THREAD

**Discussion by mingrich** [rasbt/mlxtend#919] [Q&A] [answered]
In pycharm, the data is transcoded according to the official code, and the title is also transcoded.
在pycharm中按照官方给的代码对数据进行转码，标题也被转码了。
This is the original data：
这个是原数据
![image](https://user-images.githubusercontent.com/104715355/166140977-e689d45b-2363-4971-b45b-dda08a6f80f9.png)
This is the transcoded data：
这个是转码后的数据

![image](https://user-images.githubusercontent.com/104715355/166140891-0fbd3a47-7977-41df-b334-76cc0c97ada2.png)
Excuse me, how can I make the transcoded data show my original title?
请问，我怎样可以让转码后的数据显示我原来的标题呢？

**Comment by rasbt** [comment, marked as answer]
I think the problem could be that the inputs were not lists. 

The dataset should be in the following form

```python
dataset = [['Milk', 'Onion', 'Nutmeg', 'Kidney Beans', 'Eggs', 'Yogurt'],
           ['Dill', 'Onion', 'Nutmeg', 'Kidney Beans', 'Eggs', 'Yogurt'],
           ['Milk', 'Apple', 'Kidney Beans', 'Eggs'],
           ['Milk', 'Unicorn', 'Corn', 'Kidney Beans', 'Yogurt'],
           ['Corn', 'Onion', 'Onion', 'Kidney Beans', 'Ice cream', 'Eggs']]

import pandas as pd
from mlxtend.preprocessing import TransactionEncoder

te = TransactionEncoder()
te_ary = te.fit(dataset).transform(dataset)
df = pd.DataFrame(te_ary, columns=te.columns_)

```

If you don't use sublists, I think that problems that you are showing can happen.
